### DISCUSSION TITLE
Is it possible to include CSS file internally within a <style> tag in the output HTML file ?

### DISCUSSION THREAD

**Discussion by TonyNguyen137** [webpack/webpack#18680] [Q&A] [answered]
this is my setting:

/src
|- index.html
|- critical.scss
|- non-critical.scss
|- index.js

my desired HTML output in /dist/index.html:

```
<html>
  <head>
  ...
    <link rel="stylesheet" href="non-critical.css"></link>
    <style>
     **css code from critical.scss**
    </style>
  </head>
  <body>
    ...
  </body>
</html>
```

 I'm currently using style-loader and MiniCssExtractPlugin in my Webpack setup. I understand that style-loader injects CSS into the DOM via JavaScript, while MiniCssExtractPlugin extracts the CSS into a separate file. Is there a way to achieve a different setup where my desired SCSS File  is directly included within the <style> tags in the HTML without relying on JavaScript.

here is my repo: https://github.com/TonyNguyen137/webpack-test

**Comment by alexander-akait** [comment]
You can use https://github.com/dustinjackson/html-webpack-inline-source-plugin, anyway we planned to integrate it into html-webpack-plugin

**Reply by TonyNguyen137** [reply]
Thank you, Alexander, for taking the time to respond. Unfortunately, the plugin isn't working, likely because it hasn't been maintained for the past 6 years. How can I find out when this feature will be added to Webpack? Should I follow your page or the Webpack page, or is there another way to receive notifications?

**Reply by alexander-akait** [reply, marked as answer]
https://github.com/jantimon/html-webpack-plugin/issues/1831, there is only about `scrit`, but I am fine with `style` too, so pr welcome, should be easy to implement
